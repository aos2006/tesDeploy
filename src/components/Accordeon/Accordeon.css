.root {
  width: 100%;
}

.acordeonHeader {
  display: flex;
  padding: 12px;
  align-items: center;
  justify-content: space-between;
  min-height: 34px;
  padding: 0px 20px 0px 20px;
  background-color: transparent;
  transition: background-color 0.35s cubic-bezier(0.4, 0, 0.2, 1);
  cursor: pointer;
  font-size: 1.1rem;
  font-weight: 400;
  text-transform: uppercase;
  letter-spacing: 0.047rem;
  color: rgb(192, 192, 192);
  border-bottom: 1px solid rgb(239, 239, 239);
}

.arrow {
  transform: none;
  transition: transform 0.2s;
  width: 2rem;
  fill: rgb(192, 192, 192);
}

.activeArrow {
  transform: rotate(180deg);
  transition: transform 0.2s;
}


.acordeonHeader:hover {
  background-color: #eee;
}

.accordeonContent {
  max-height: 0;
  padding-top: 0;
  padding-bottom: 0;
  overflow: hidden;
  height: 0;
  transition: max-height 0.3s;
  will-change: max-height;
}

.acordeonActiveHeader {
  background-color: rgb(239, 239, 239); 
}

.accordeonActiveContent {
  max-height: 500px;
  transition: max-height 0.3s;
  height: auto;
  border-bottom: 1px solid rgb(239, 239, 239);
}

