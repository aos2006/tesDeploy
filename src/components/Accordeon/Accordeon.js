import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import cx from 'classnames';
import s from './Accordeon.css';

const Accordeon = props => (
  <div className={cx({
    [s.root]: true,
    [props.classnames.root]: true,
  })}>
    {React.Children.map(props.children, (child, index) =>
      React.cloneElement(child, {
        active: index === props.activeIndex,
        handleToggle: props.onOpen,
        itemIndex: index,
        classnames: props.classnames,
      }),
    )}
  </div>
);

Accordeon.propTypes = {
  activeIndex: PropTypes.number.isRequired,
  onOpen: PropTypes.func.isRequired,
  children: PropTypes.any.isRequired,
};


Accordeon.defaultProps = {
  classnames: {
    root: '',
  }
}

export default withStyles(s)(Accordeon);
