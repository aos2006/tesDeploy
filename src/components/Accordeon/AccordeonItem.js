import React from 'react';
import PropTypes from 'prop-types';
import FontIcon from 'react-toolbox/lib/font_icon';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import Arrow from '-!babel-loader!svg-react-loader!./icons/arrow.svg?name=Arrow';
import cx from 'classnames';
import s from './Accordeon.css';

const handleToggle = (index, fn = () => {}) => () => fn(index);

const AccordeonItem = props => (
  <div>
    <header
      className={cx({
        [s.acordeonHeader]: true,
        [s.acordeonActiveHeader]: props.active,
      })}
      onClick={handleToggle(props.itemIndex, props.handleToggle)}
    >
      <Arrow
        className={cx({
          [s.arrow]: true,
          [s.activeArrow]: props.active,
        })}
      />

      {props.title}
    </header>
    <div
      className={cx({
        [s.accordeonContent]: true,
        [s.accordeonActiveContent]: props.active,
        [props.classnames.content]: true,
        [props.classnames.activeContent]: props.active,
      })}
    >
      {props.children}
    </div>
  </div>
);

AccordeonItem.propTypes = {
  active: PropTypes.bool.isRequired,
  itemIndex: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  handleToggle: PropTypes.func.isRequired,
  children: PropTypes.any.isRequired,
};

export default withStyles(s)(AccordeonItem);
