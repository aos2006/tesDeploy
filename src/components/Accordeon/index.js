export { default as Accordeon } from './Accordeon';
export { default as AccordeonItem } from './AccordeonItem';
