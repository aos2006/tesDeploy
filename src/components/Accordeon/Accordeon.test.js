import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordeon from './Accordeon';
import AccordeonItem from './AccordeonItem';

describe('<Accordeon />', () => {
  const handleOpen = jest.fn();
  const list = [
    {
      title: 'Меню 1',
    },
    {
      title: 'Меню 2',
    },
  ];

  it('Render testing', () => {
    const wrapper = shallow(
      <Accordeon activeIndex={0} onOpen={handleOpen}>
        {list.map(item => (
          <AccordeonItem title={item.title}>dimon bidon</AccordeonItem>
        ))}
      </Accordeon>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('When pass index item:index is open', () => {
    const wrapper = shallow(
      <Accordeon activeIndex={0} onOpen={handleOpen}>
        {list.map(item => (
          <AccordeonItem title={item.title}>dimon bidon</AccordeonItem>
        ))}
      </Accordeon>,
    );
    const ACTIVE_LENGTH = wrapper.find('[active=true]').length;
    expect(ACTIVE_LENGTH).toEqual(1);
  });
});
