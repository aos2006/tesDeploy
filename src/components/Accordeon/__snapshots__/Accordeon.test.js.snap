// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordeon /> Render testing 1`] = `
<div
  className="root"
>
  <AccordeonItem
    active={true}
    handleToggle={[Function]}
    itemIndex={0}
    title="Меню 1"
  >
    dimon bidon
  </AccordeonItem>
  <AccordeonItem
    active={false}
    handleToggle={[Function]}
    itemIndex={1}
    title="Меню 2"
  >
    dimon bidon
  </AccordeonItem>
</div>
`;
